pir/cpp/client.cpp

```diff
@@ -41,69 +46,58 @@ StatusOr<std::unique_ptr<PIRClient>> PIRClient::Create(
 }
 
 StatusOr<PIRPayload> PIRClient::CreateRequest(std::size_t index) const {
-  if (index >= this->DBSize()) {
+  const auto poly_modulus_degree =
+      context_->Parameters()->GetEncryptionParams().poly_modulus_degree();
+  if (index >= DBSize()) {
     return InvalidArgumentError("invalid index");
   }
-  std::vector<std::int64_t> request(DBSize(), 0);
-  request[index] = 1;
-
-  return this->encryptRequestBuffer(request);
+  if (index >= poly_modulus_degree) {
+    // Not yet implemented
+    return InvalidArgumentError("More than 1 CT needed for selection vector");
+  }
+  Plaintext pt(poly_modulus_degree);
+  pt.set_zero();
+  pt[index] = 1;
+  std::cout << "Query PT: " << pt.to_string() << std::endl;
+
+  vector<Ciphertext> query(1);
+  GaloisKeys gal_keys;
+  try {
+    encryptor_->encrypt(pt, query[0]);
+    gal_keys =
```

Comment:
Note that there could be a reason not to perform full compression (i.e., do less than logN levels during the oblivious expansion), in which case less Galois keys would be necessary. Even if it's not in this PR, you may want to think about how to make the code modular to handle this.


You should only send the Galois keys necessary

Reply:
Interesting... I'd like to discuss when it would be worthwhile not doing the full expansion. For now, I've created https://github.com/OpenMined/PIR/issues/8 . Thanks for this!